csrc/python_frontend/fusion_definition.cpp

```diff
@@ -403,6 +404,10 @@ State FusionDefinition::recordingState(size_t index) const {
   return recording_state_.at(index);
 }
 
+const std::vector<Tensor>& FusionDefinition::getAllTensors() const {
+  return all_tensors_;
```

Comment:
I would strongly encourage against creating `all_tensors_` container.

You should be able to just filter recording_state_ to find indices to `TensorView` entries in `fusion_state_`. We can then construct `Tensor` with the `Tensor(index, tv->nDims(), this)`.